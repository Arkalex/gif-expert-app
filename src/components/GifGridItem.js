import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {
    return (
        <div className="card animate_animated animate_fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
