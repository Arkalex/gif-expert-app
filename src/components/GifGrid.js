import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect( ()=> {
        getGifs(category)
            .then(imgs => setImages(imgs))
    }, [category])

    

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
            {
                images.map((img) => {
                    return <GifGridItem 
                                key={ img.id }
                                { ...img } 
                            />
                })
            }
            </div>
        </>
    )
}
