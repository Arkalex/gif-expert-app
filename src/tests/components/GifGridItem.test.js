import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem /> ', () => {

    const title = 'Titulo random';
    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow(<GifGridItem 
                                title={title} 
                                url={url} 
                            />);

    test('should show the component', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('should has a <p> with title', () => {
        
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    })

    test('should have an image with alt and url from props', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );
    })

    test('should have class animate_fadeIn', () => {

        const div = wrapper.find('div');

        const className = div.prop('className');
        expect( className.includes('animate_fadeIn') ).toBe( true );
        
    })
    
})