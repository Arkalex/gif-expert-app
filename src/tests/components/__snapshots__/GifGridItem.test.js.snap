// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem />  should show the component 1`] = `
<div
  className="card animate_animated animate_fadeIn"
>
  <img
    alt="Titulo random"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo random
  </p>
</div>
`;
